extensions/rocksdb-repos/DatabaseContentRepository.cpp

```diff
@@ -72,13 +87,40 @@ bool DatabaseContentRepository::initialize(const std::shared_ptr<minifi::Configu
   return is_valid_;
 }
 
+void DatabaseContentRepository::runCompaction() {
+  do {
+    if (auto opendb = db_->open()) {
+      auto status = opendb->RunCompaction();
+      logger_->log_trace("Compaction triggered: %s", status.ToString());
+    } else {
+      logger_->log_error("Failed to open database for compaction");
+    }
+  } while (!utils::StoppableThread::wait_stop_requested(compaction_period_));
+}
+
+void DatabaseContentRepository::start() {
+  if (!db_ || !is_valid_) {
+    return;
+  }
+  if (compaction_period_.count() != 0) {
+    compaction_thread_ = std::make_unique<utils::StoppableThread>([this] () {
+      runCompaction();
+    });
+  }
+}
+
 void DatabaseContentRepository::stop() {
   if (db_) {
     auto opendb = db_->open();
     if (opendb) {
       opendb->FlushWAL(true);
     }
+    compaction_thread_.reset();
```

Comment:
I think this should block until the thread is stopped, to avoid accessing resources that are being cleaned up.

Reply:
the thread's destructor called during reset block until it is joined